lib/filesystem/src/filesystem.cpp

```diff
@@ -160,17 +160,31 @@ namespace irods::experimental::filesystem::NAMESPACE_IMPL
             else if (COLL_OBJ_T == _s.type) {
                 irods::experimental::query_builder qb;
 
-                std::vector<std::string> args{_p.string()};
-
-                qb.type(irods::experimental::query_type::specific)
-                  .bind_arguments(args);
-
                 if (const auto zone = get_zone_name(_p); zone) {
                     qb.zone_hint(*zone);
                 }
 
-                for (const auto& row : qb.build(_comm, "ShowCollAcls")) {
-                    _s.prms.push_back({row[0], to_permission_enum(row[2])});
+                try {
+                    std::vector<std::string> args{_p.string()};
+
+                    qb.type(irods::experimental::query_type::specific)
+                      .bind_arguments(args);
+
+                    for (const auto& row : qb.build(_comm, "ShowCollAcls")) {
+                        _s.prms.push_back({row[0], to_permission_enum(row[2])});
+                    }
+                }
+                catch (...) {
```

Comment:
can we / do we want to avoid the (...)?  a more specific exception to catch?

Reply:
I thought about that and ultimately decided it wasn't needed. The current fsys impl does not log anything. In nearly every case, it throws an exception for the user to handle. With that, I didn't see any benefit to specifying a type if it was not going to be used.

Reply:
cool.   maybe add a comment to save that thinking/why.

# away.

Reply:
Sure. After reviewing this a bit more, I don't understand why ils succeeds with this change. I understand the fsys fix, but I would expect ils to still fail given that iquest fails to link tables when executing the same genquery string used by the following function:  https://github.com/irods/irods/blob/0a31d045e43396e198b7a127419dc30d60df3151/lib/core/src/lsUtil.cpp#L666

Reply:
It turns out that `queryCollAcl` uses columns that are not mapped. Because of that, `fillGenQueryInpFromStrCond` cannot find the required columns when attempting to generate the correct SQL.

Created issues and I also have fixes for this included in this PR as of now.

Reply:
very nice - remarkable that those columns weren't mapped.    they now appear in `iquest attrs` and hadn't before?

Reply:
confirming they didn't in 4.2.7...

```
$ imiscsvrinfo | grep relVersion
relVersion=rods4.2.7
$ iquest attrs | grep COLL_USER_NAME
$ echo $?
1
$ iquest attrs | grep COLL_ZONE_NAME
$ echo $?
1
```

Reply:
Yes, they do appear in `iquest attrs`.